lib/jekyll/document.rb

```diff
@@ -433,15 +353,129 @@ def respond_to?(method, include_private = false)
     # Override of method_missing to check in @data for the key.
     def method_missing(method, *args, &blck)
       if data.key?(method.to_s)
-        Jekyll.logger.warn "Deprecation:", "Document##{method} is now a key in the #data hash."
+        Jekyll.logger.warn(
+          "Deprecation:",
+          "Document##{method} is now a key in the #data hash."
+        )
```

Comment:
Let's do `Jekyll::Deprecator. deprecation_message "Document##{method} is now a key in the #data hash."`